src/main/java/mate/academy/dao/MovieSessionDao.java

```diff
@@ -0,0 +1,16 @@
+package mate.academy.dao;
+
+import java.time.LocalDate;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.model.MovieSession;
+
+public interface MovieSessionDao {
+    MovieSession add(MovieSession movieSession);
+
+    Optional<MovieSession> get(Long id);
+
+    List<MovieSession> findAvailableSessions(Long movieId, LocalDate date);
+
+    List<MovieSession> findAll();
```

Comment:
According to the checklist, we should not create redundant functionality. The method `findAll()` is not required by the task and should be removed.